lib/vyasa/written.ex

```diff
@@ -79,9 +124,19 @@ defmodule Vyasa.Written do
   end
 
   def get_chapter(no, source_title) do
-    src = get_source_by_title(source_title)
-    Repo.get_by(Chapter, no: no, source_id: src.id)
-    |> Repo.preload([:translations, verses: [:translations]])
+    (from c in Chapter, where: c.no ==  ^no,
+      inner_join: src in assoc(c, :source),
+      where: src.title == ^source_title)
+    |> Repo.one()
+  end
+
+  def get_chapter(no, source_title, lang) do
+    target_lang = (from ts in Translation, where: ts.lang == ^lang)
+    (from c in Chapter, where: c.no ==  ^no,
+      inner_join: src in assoc(c, :source),
+      where: src.title == ^source_title,
+      preload: [verses: ^(from v in Verse, preload: [translations: ^target_lang]) , translations: ^target_lang])
```

Comment:
it allows associate chap <-> verse to have the same translations


the == lang can be extended to support a list of translations in case people want to have mutiple trnaslations here